expression/builtin_cast.go

```diff
@@ -1294,9 +1294,6 @@ func WrapWithCastAsReal(expr Expression, ctx context.Context) (Expression, error
 	}
 	tp := types.NewFieldType(mysql.TypeDouble)
 	tp.Flen, tp.Decimal = mysql.MaxRealWidth, types.UnspecifiedLength
-	if exprTp := expr.GetType(); exprTp.ToClass() == types.ClassInt {
```

Comment:
how about add a `TODO` to indicate we should revert this change after all functions being rewrited ?